src/petals/server/server.py

```diff
@@ -220,11 +220,10 @@ def __init__(
             num_blocks = min(num_blocks, self.block_config.num_hidden_layers)
         if block_indices is not None:
             try:
-                first_block_index, last_block_index = block_indices.split(":")
-                first_block_index, last_block_index = map(int, map(str.strip, (first_block_index, last_block_index)))
```

Comment:
It's not actually the "last" block, but the one after last. Also, this code can be written in a simpler way